### DISCUSSION TITLE
module request: expiration tracking: domains & certificates

### DISCUSSION THREAD

**Discussion by SpamFaux** [blacklanternsecurity/bbot#720] [Ideas]
**Description**
Which feature would you like to see added to BBOT? What are its use cases?

As BBOT already collects some data from Certificates it would be useful to track the certificates themselves for monitoring Valid Through dates. 

Additionally, being able to note if domains are soon to expired - this can highlight potential issues / risks to availability of a target's domain/subdomains.

**Comment by aconite33** [comment]
Domain will be manually
Certificates are doable
When raised as event, compare expiration with current time
Annotate tag with expiring in X days
Domain will have to be registrar, future work would require additional service to review whois data

**Comment by Sh4d0wHunt3rX** [comment]
That would be great if you can add to the asset inventory file, including expired, revoked, wildcard, self signed, mismatched

Thanks 🙏

**Comment by liquidsec** [comment]
@SpamFaux is this something you are still interested in / still a gap for you?

**Reply by SpamFaux** [reply]
Something to this effect would still be useful. Many companies and some of our clients struggle with certificate expiration management. Being able to track that for clients in BBOT would be a net positive. Even if it is just grabbing the valid thru date.
